refgenconf/refgenconf.py

```diff
@@ -2087,11 +2166,154 @@ def _select_genomes(self, genome=None, strict=False, order=None,
+            my_genome.update([(v[CFG_ALIASES_KEY][0], k)])
+
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)))
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR)))
+
+        # copy folder
+        for root, dirs, files in os.walk(rgc[CFG_FOLDER_KEY]):
+            for dir in dirs:
+                if dir in my_genome:
+                    shutil.copytree(os.path.join(rgc[CFG_FOLDER_KEY], dir),
+                                    os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR, dir))
+            del dirs[:]
+
+        for root, dirs, files in os.walk(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)):
+            for dir in dirs:
+                _swap_names_in_tree(os.path.join(
+                    root, dir), my_genome[dir], dir)
+                os.mkdir(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR, dir))
+                # create symlink for alias folder
+                for genome, assets, files in os.walk(os.path.join(root, my_genome[dir])):
+                    for asset in assets:
+                        old_path = os.path.join(genome, asset)
+                        new_path = old_path.replace(
+                            my_genome[dir], dir).replace(DATA_DIR, ALIAS_DIR)
+                        os.mkdir(new_path)
+                    for file in files:
+                        old_path = os.path.join(genome, file)
+                        new_path = old_path.replace(
+                            my_genome[dir], dir).replace(DATA_DIR, ALIAS_DIR)
+                        link_fun(old_path, new_path)
+            del dirs[:]
+
+        for genome, genome_v in rgc[CFG_GENOMES_KEY].items():
+            d = os.path.join(rgc[CFG_FOLDER_KEY], genome_v[CFG_ALIASES_KEY][0])
+            shutil.rmtree(d)
+
+    rgc = _RefGenConfV03(filepath=filepath, writable=True)
+    # prompt the user
+    # TODO: add upgrade-specific docs page
+    url = "http://refgenie.databio.org/" 
+    if not force and not query_yes_no(
+            f"Upgrading config to v{target_version}. Current genome identifiers "\
+            f"will be replaced with sequence-derived digests and contents "\
+            f"inside {rgc[CFG_FOLDER_KEY]}' will be replaced by '{DATA_DIR}' "\
+            f"and '{ALIAS_DIR}' directories. For more info visit: {url}. "\
+            f"Would you like to proceed?"):
+        _LOGGER.info("Action aborted by the user.")
+        return
+
+    missing_digest = format_config(target_version)  # reformat config file
+    if not force and missing_digest and not query_yes_no(
+        f"The following genomes will be lost due to the lack of local fasta "\
+        f"assets and remote genome digests: {', '.join(missing_digest)}. "\
+        f"Would you like to proceed?"):
+        _LOGGER.info("Action aborted by the user.")
+        return
+
+    # change the config_version
+    rgc[CFG_VERSION_KEY] = target_version
+    # write over the config file
+    rgc.write()
+
+    alter_file_tree()
```

Comment:
I think the config should be overwritten after successful file tree update. Now, if the file tree update fails, we end up with an updated config and old file tree structure, which are incompatible.

Reply:
did this get resolved?